controllers/pwa.js

```diff
@@ -22,51 +22,46 @@ const lighthouseLib = require('../lib/lighthouse');
 const Pwa = require('../models/pwa');
 const router = express.Router(); // eslint-disable-line new-cap
 const libMetadata = require('../lib/metadata');
+const cache = require('../lib/data-cache');
 
 const LIST_PAGE_SIZE = 32;
 const DEFAULT_PAGE_NUMBER = 1;
 const DEFAULT_SORT_ORDER = 'newest';
 
+const CACHE_LIFETIME = 60 * 60 * 24; // 24 hours
+
 /**
  * GET /
  *
- * Display a page of PWAs (up to ten at a time).
+ * Display a page of PWAs (up to LIST_PAGE_SIZE at a time)
+ * Stores the latest page in memcached.
  */
 router.get('/', (req, res, next) => {
-  const pageNumber = parseInt(req.query.page, 10) || DEFAULT_PAGE_NUMBER;
-  const sortOrder = req.query.sort || DEFAULT_SORT_ORDER;
-  const start = parseInt(req.query.start, 10) || (pageNumber - 1) * LIST_PAGE_SIZE;
-  const limit = parseInt(req.query.limit, 10) || LIST_PAGE_SIZE;
-  const end = pageNumber * LIST_PAGE_SIZE;
-  const contentOnly = false || req.query.contentOnly;
-  let pwaCount = 0;
-  pwaLib.count()
-    .then(count => {
-      pwaCount = count;
-      return pwaLib.list(start, limit, sortOrder);
+  cache.get(req.originalUrl)
```

Comment:
We may create the rendered page cache layer as a middleware. When creating the middleware, we can define which routes we want this middleware to handle. This would be a more generic approach and would leave the PWA controller to only care about rendering pages that are not cached.